.github/workflows/ci-cd.yml

```diff
@@ -135,55 +65,20 @@ jobs:
               with:
                 name: dist-output
                 path: ./dist
-    test-integration:
-        needs: build
-        runs-on: ubuntu-latest
-        env:
-            JEST_JUNIT_OUTPUT_NAME: results.txt
-            JEST_JUNIT_OUTPUT_DIR: test-results/integration
-        steps:
-            - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
-            - name: Retrieve npm dependencies
-              uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3
-              with:
-                path:
-                    node_modules
-                key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json') }}
-            - name: Retrieve Build
-              uses: actions/cache@e12d46a63a90f2fae62d114769bbf2a179198b5c # v3
-              with:
-                path:
-                    ./build
-                key: ${{ runner.os }}-build-${{ hashFiles('package-lock.json') }}
             - run: |
                 for F in chrome chromium chromedriver; do
                     which $F && $F --version || echo Not found: $F
                 done
             - name: Run Integration Tests
+              env:
+                  JEST_JUNIT_OUTPUT_NAME: results.txt
+                  JEST_JUNIT_OUTPUT_DIR: test-results/integration
               run: npm run test:integration -- --reporters="default" --reporters="jest-junit"
             - name: Store Integration Test Results
               uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3
               with:
                 name: integration-test-output
                 path: ./test-results/*
```

Comment:
With the the env variables as they are the integration tests artifact would contain both unit and integration tests result.

Reply:
Ah, good point - thanks!

Reply:
I combined the two steps for storing test results, so now there's one artifact that contains any/all output from both testing steps. I set it to run "always" so that (for example) if the unit tests have output but the build fails, the unit test output is still available. That also means any output from failed tests will be available, which probably should have been the way we set it up to begin with!